tests/ServerRequestTest.php

```diff
@@ -519,4 +519,19 @@ public function testNullAttribute()
         $this->assertSame([], $requestWithoutAttribute->getAttributes());
         $this->assertSame('different-default', $requestWithoutAttribute->getAttribute('name', 'different-default'));
     }
+    
+    public function testExtend()
+    {
+        $actual = FooExtend::fromGlobals();
+        $this->assertInstanceOf('GuzzleHttp\\Tests\\Psr7\\FooExtend', $actual);
+        $this->assertTrue($actual->bar());
+    }
+}
+
+class FooExtend extends ServerRequest
```

Comment:
@ilyar Thanks for creating this fix, and thanks especially for creating a test case, which is something I forgot to do in mine :).

Just wanted to share an opinion: To me, it seems better to define `FooExtend` in a separate file. I often do this by using a `classes` folder under `tests`, namespacing those classes with `Test`, and then including them in the `autoload-dev` section of the composer file. The modified `composer.json` might look like this:

```json
    ....
        "files": ["src/functions_include.php"]
    },
    "autoload-dev": {
        "psr-4": {
            "\\Test": "tests/classes/"
        }
    },
    "extra": {
    ....
```

Then you can use the class in any test by calling `\Test\FooExtend`.

Anyway, take it or leave it!

Reply:
@kael-shipman Maybe it would be appropriate to call it so `GuzzleHttp\Tests\Fixture\FooExtend`?

Reply:
That could work, though I always thought of Fixtures as predictable data states (i.e., NOT just classes meant to test functionality of other classes).

My vote is to make the namespace `\GuzzleHttp\Test`. That would keep it simple and also allow them to possibly break that set of test classes out into a separate repo if they start using them across multiple sub-projects (something I do occasionally).

Anyway, your call!

Reply:
fix